src/builtin_string.cpp

```diff
@@ -1327,8 +1327,8 @@ static int string_trim(parser_t &parser, io_streams_t &streams, int argc, wchar_
     return ntrim > 0 ? STATUS_CMD_OK : STATUS_CMD_ERROR;
 }
 
-/// Implementation of `string lower`.
-static int string_lower(parser_t &parser, io_streams_t &streams, int argc, wchar_t **argv) {
+// A helper function for lower and upper.
+static int string_transform(parser_t &parser, io_streams_t &streams, int argc, wchar_t **argv, decltype(std::towlower) func) {
```

Comment:
`decltype(std::towlower)` is some fancy!

Reply:
TBH I was simply too lazy to write out the full type :smile:.